addons/web/static/src/core/l10n/translation.js

```diff
@@ -78,3 +97,45 @@ export async function loadLanguages(orm) {
     }
     return loadLanguages.installedLanguages;
 }
+
+/**
+ * Same behavior as sprintf, but if any of the provided values is a markup,
+ * escapes all non-markup content before performing the interpolation, then
+ * wraps the result in a markup.
+ *
+ * @param {string} str The string with placeholders (%s) to insert values into.
+ * @param  {...any} values Primitive values to insert in place of placeholders.
+ * @returns {string|Markup}
+ */
+function _safeSprintf(str, ...values) {
+    let hasMarkup;
+    if (values.length === 1 && Object.prototype.toString.call(values[0]) === "[object Object]") {
+        hasMarkup = Object.values(values[0]).some((v) => v instanceof Markup);
+    } else {
+        hasMarkup = values.some((v) => v instanceof Markup);
+    }
+    if (hasMarkup) {
+        return markup(sprintf(escape(str), ..._escapeNonMarkup(values)));
+    }
+    return sprintf(str, ...values);
+}
+
+/**
+ * Go through each value to be passed to sprintf and escape anything that isn't
+ * a markup.
+ *
+ * @param {any[]|[Object]} values Values for use with sprintf.
+ * @returns {any[]|[Object]}
+ */
+function _escapeNonMarkup(values) {
+    if (Object.prototype.toString.call(values[0]) === "[object Object]") {
```

Comment:
the following code is 1 line shorter, does not create a bunch of arrays that will be garbage collected, and is simpler. i like smart, but i do not see the advantage here
```js
const result = {};
for (const k in values[0]) {
  const v = values[0][k];
  result[k] = v instanceof Markup ? v : escape(v);
}
return result;
```

Reply:
Agreed, I had an idea in mind and got tunnel vision, which led to this piece of code. I changed it, thanks!